File: 字符串/例题/LC1071-字符串的最大公因子/1071.py
import math
class Solution:
	def gcdOfStrings(self, str1: str, str2: str) -> str:
		my = math.gcd(len(str1), len(str2))
		stm = str1[:my]
		if stm * (len(str1)//my) == str1 and stm * (len(str2)//my) == str2:
			return stm
		return ""
if __name__ == "__main__":
	a = Solution()
	print(a.gcdOfStrings("ABCABCABC","ABCABC"))

File: 字符串/例题/LC1071-字符串的最大公因子/1071.cpp
#include <iostream>
#include <numeric>
using namespace std;
class Solution
{
private:
	/* data */
public:
	Solution(/* args */){}
	~Solution(){}
	bool issub(string sub, string a) {
		int bl = (int)a.length()/(int)sub.length();
		string s = "";
		for(int i = 1; i <= bl; i++)		s += sub;
		return s == a;	
	}
	string gcdOfStrings(string str1, string str2) {
		int l1 = (int)str1.length(), l2 = (int)str2.length();
		string s = str1.substr(0, gcd(l1, l2));
		if (issub(s, str1) && issub(s, str2))	return s;
		return "";
	}
};
int main() {
	Solution a;
	cout << a.gcdOfStrings("ABCABCABC","ABCABC");
	return 0;
}